src/JuMP.jl

```diff
@@ -525,6 +529,16 @@ function getDual(c::ConstraintRef{LinearConstraint})
     return c.m.linconstrDuals[c.idx]
 end
 
+function getDual(c::ConstraintRef{SOCConstraint})
+    #=if length(c.m.linconstrDuals) != MathProgBase.numsocconstr(c.m)
+        error("Dual solution not available. Check that the model was properly solved and no integer variables are present.")
+    end=#
+    return c.m.conicconstrDuals[
+        c.m.constrDualMap[length(c.m.linconstr) + c.idx]]
```

Comment:
``length(c.m.linconstr)`` is not necessarily constant. This will give junk if users add a constraint after solving the model and then ask for a dual from the previous solution. It's okay to return an error in this case, but not junk.